### DISCUSSION TITLE
Pass Query Param without encoding colon to %3A

### DISCUSSION THREAD

**Discussion by dktcl** [grafana/grafana-infinity-datasource#194] [Q&A]
Hi,

I am trying to pass Grafana Global Variable for timestamp in IOS format (e.g. `YYYY-MM-DDThh:mm:ssZ`), however when I assign a Grafana Global Variable (e.g. `${__from:date}` or `${__from:date:iso}`), the URL called by Infinity encodes the colons as %3A....

Is it possible to disable  encoding of colons?

Thanks,

Dave

**Comment by yesoreyeram** [comment]
See if this helps alternatively https://github.com/yesoreyeram/grafana-infinity-datasource/discussions/38#discussioncomment-450607

**Comment by dktcl** [comment]
Hi @yesoreyeram - thanks for your reply (and your great work building this plugin!)

I should have said that I tried #38, using both
`${__to:date:YYYY-MM-DD}%20${__to:date:HH}:${__to:date:mm}:${__to:date:ss}` 
AND
`end=${__to:date:YYYY-MM-DD}T${__to:date:HH}:${__to:date:mm}:${__to:date:ss}Z`

However, the issue persists. I've tried using Console on Chrome's Developer Tools which suggests that the request from Grafana to `url: "api/datasources/6/resources/proxy"` is formatted "correctly" (see below).

![image](https://user-images.githubusercontent.com/36790933/140479031-91433afc-577d-4f94-b074-5bf42890032f.png)

However, when I check my API endpoint, it receives this:
`/v1/qry/table/states?device_id=3&ts_from=2021-11-05T02%3A14%3A11Z&ts_to=2021-11-05T08%3A14%3A11Z`

I've tested my endpoint from Postman, manual calls from a python script, NodeRed, etc - and all is OK.

Infinity 0.7.8 on Grafana 8.1.3 running on Docker on a Debian 10 host.

Thanks for your help. If you want me to create a bug report, I can.

**Reply by dktcl** [reply]
Hi @yesoreyeram - would you like me to file a bug report instead? Thanks

**Reply by yesoreyeram** [reply]
no worries. #198 created. Should be easy to fix. But wonder what is the expected behaviour. If we change the behaviour, this may break for many other users. So need some solid evidence that we need to do this.

**Reply by dktcl** [reply]
Thanks.

Weirdly, I had this working in an older version of Infinity and Grafana 8.x... i.e. ISO string was not escaped when it reached our endpoints.

To be fair, opinion seems to be divided online regards escaping or not escaping. It's certainly a lot easier to use unescaped colons, as its the default in Grafana, and anywhere else you use `.toISOString()`.

A toggle to select escape or not is most user friendly, but probably priority for you.

In the meantime, is there any workaround with Infinity? Otherwise, I'll need to start modifying endpoints to accept escaped and unescaped colons (Gah!!), as I would like to stick with Infinity...

Have a nice weekend
